src/query/api/v1/options/handler.go

```diff
@@ -60,6 +60,10 @@ const (
 	M3QueryEngine QueryEngine = "m3query"
 )
 
+// PromQLEngineFn construct promql.Engine with the given lookbackDuration. promql.Engine uses
+// a fixed lookback, so we have to create multiple engines for different lookback values.
```

Comment:
For better tracking, I would add a TODO with link to the Prometheus mailing list (or any more appropriate resource).

Reply:
https://github.com/m3db/m3/pull/3882/commits/5901ce5b838f3dadda39230774a7edfc6be566b7#diff-0d8ef3d8555895061a7eb7f4b1e05852283ade888692425afe6db8cad09a4e17R66-R67